genomepy/caching.py

```diff
@@ -1,26 +1,87 @@
 import os
+import time
 from shutil import rmtree
+from functools import wraps
+from threading import Lock
 
 from appdirs import user_cache_dir
 from bucketcache import Bucket
-from joblib import Memory
+from joblib import Memory,dump,load,hash
+from diskcache import Cache
 
 from genomepy.__about__ import __version__
 
 genomepy_cache_dir = os.path.join(user_cache_dir("genomepy"), __version__)
 os.makedirs(genomepy_cache_dir, exist_ok=True)
-
 # Store the output of slow commands (marked with @cache and @goldfish_cache) for fast reuse.
 # Bucketcache creates a new pickle for each function + set of unique variables,
 # For class methods, use ignore=["self"] or use @staticmethod.
 cache = Bucket(genomepy_cache_dir, days=7)
 goldfish_cache = Bucket(genomepy_cache_dir, minutes=10)
 
 memory = Memory(genomepy_cache_dir, verbose=0)
-
+disk_cache = Cache(directory=genomepy_cache_dir)
 
 def clean():
     """Remove cached data on providers."""
     rmtree(genomepy_cache_dir, ignore_errors=True)
     os.makedirs(genomepy_cache_dir, exist_ok=True)
     print("All clean!")
+
+def hybcache(exp_short=3600, exp_long=3600, cache=disk_cache, lock=Lock()):
+    def decorator(func):
+        @wraps(func)
+        def wrapper(*args, **kwargs):
+            # Generate the cache key from the function's arguments.
+            print(args[0].__dict__)
+            key = [func.__name__] + list(args) + list(sorted(kwargs.items()))
```

Comment:
Why are the (**kwargs items) sorted first?

Reply:
Are they stored in a non-orderd dictionary? Aaah i think i get it!

Reply:
The sorting is actually a bad thing I think. You shuold sort on the keys, otherwise these two calls give the same result:

```
pow(2, 5)
```


```
pow(5, 2)
```

:ghost: